### DISCUSSION TITLE
Array of component resistor on OpenModelica

### DISCUSSION THREAD

**Discussion by bflo96** [OpenModelica/OpenModelica#12255] [Q&A] [answered]
Hello everyone,

I've recently started to develop a simple model on OpenModelica (3 days ago), which consists on several resistances connected to one another and 1 current source. I know _a priori_ the value of each resistance will be characterized by a polynomial expression of 3rd degree, but I started by doing a very simple exercise to learn how to relate everything before jumping to a more complicated schematic.

![image](https://github.com/OpenModelica/OpenModelica/assets/167191747/a4ebe8fd-478f-4710-b26b-c330dd8ef2ef)

While for this case, I have a fixed number of resistances, I want to be able to have multiple resistances in parallel.
This is a snippet of my code (connectors are hidden). I can't compile because it gives 54 equations and 50 variables,
![image](https://github.com/OpenModelica/OpenModelica/assets/167191747/48ee7b83-12e3-441f-9d50-c07bafad3171)

Alternatively, I tried to use a loop, but it increases the number of equations.
![image](https://github.com/OpenModelica/OpenModelica/assets/167191747/a5b2bb7f-7ae1-466d-b9f1-374498f694fd)

I think I'm making some kind of confusion on parameters and variables... Any kind of help would be really appreciated!

Thank you.

**Comment by casella** [comment, marked as answer]
Equations in the equation section are for variables, not for parameters. This explains why you get extra equations in the equation count. You need to use binding equations for parameters
```modelica
Modelica.Electrical.Analog.Basic.Resistor R[n](R = param_r);
```

**Reply by bflo96** [reply]
Thank you! I'll look into it.
